src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketOutput.cs

```diff
@@ -51,7 +51,7 @@ public class SocketOutput : ISocketOutput
         private bool _postingWrite = false;
 
         private bool _cancelled = false;
-        private int _numBytesPreCompleted = 0;
+        private long _numBytesPreCompleted = 0;
```

Comment:
One thing to watch out for here is that `long` is not guaranteed to have atomic reads and writes. Do we ever read and write this concurrently?